src/main/java/seedu/address/storage/JsonAdaptedTask.java

```diff
@@ -121,6 +124,10 @@ public Task toModelType() throws IllegalValueException {
         }
         final CompletionStatus modelCompletionStatus = new CompletionStatus(completionStatus);
 
+        //is deadline past today check
+        if (!new DatePastPredicate().test(modelDeadline)) {
```

Comment:
I think this should be allowed

Reply:
which means only invalid events should be disallowed. overdue tasks and past events should be allowed